<?xml version="1.0"?>
<launch>

    <arg name="points_in" default="/sensing/lidar/top/points"/>
    <arg name="twist_in" default="/devbot/twist" />
    <arg name="points_out" default="/sensing/lidar/top/points_undistorted" />
   
    <arg name="timing_correction" default="true" />
    <arg name="rotation_frequency" default="20" />
    <arg name="ouster_mode" default="true"/>

    <node pkg="arg_lidar_distortion_correction" name="arg_lidar_distortion_correction" type="arg_lidar_distortion_correction" output="screen">
      <remap from="points_in" to="$(arg points_in)"/>
      <remap from="twist_in" to="$(arg twist_in)" />
      <remap from="points_out" to="$(arg points_out)" /> 

      <param name="timing_correction" type="bool" value="$(arg timing_correction)"/>
      <param name="rotation_frequency" type="int" value="$(arg rotation_frequency)"/>
      <param name="ouster_mode" type="bool" value="$(arg ouster_mode)"/>
    </node>

</launch>
